constraints.pro

```diff
@@ -50,6 +50,12 @@ gen_enforced_dependency(WorkspaceCwd, DependencyIdent, 'workspace:*', 'devDepend
   % Only consider those that target something that could be a workspace
   workspace_ident(DependencyCwd, DependencyIdent).
 
+% This rule enforces that all workspaces must depend on other workspaces using `workspace:^` in peerDependencies
+gen_enforced_dependency(WorkspaceCwd, DependencyIdent, 'workspace:^', 'peerDependencies') :-
+  workspace_has_dependency(WorkspaceCwd, DependencyIdent, _, 'peerDependencies'),
+  % Only consider those that target something that could be a workspace
+  workspace_ident(DependencyCwd, DependencyIdent).
```

Comment:
We should be able to remove the two constraints above this and make this one target all dependency types

Reply:
Yep I didn't want to make the diff too large, but I'll do it as soon as it gets merged.